cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/MonitorHandlerTest.java

```diff
@@ -102,22 +102,44 @@ public void testInstances() throws Exception {
 
   @Test
   public void testRestartInstances() throws Exception {
-    String path = String.format("system/services/%s/instances/restart", Constants.Service.APP_FABRIC_HTTP);
-    HttpURLConnection urlConn = openURL(path, HttpMethod.PUT);
-
-    Assert.assertEquals(HttpResponseStatus.OK.getCode(), urlConn.getResponseCode());
+    String path = String.format("%s/system/services/%s/restart", Constants.Gateway.API_VERSION_3,
+                                Constants.Service.APP_FABRIC_HTTP);
+    HttpResponse response = doPost(path);
 
-    urlConn.disconnect();
+    Assert.assertEquals(HttpResponseStatus.OK.getCode(), response.getStatusLine().getStatusCode());
 
-    urlConn = openURL(path, HttpMethod.GET);
+    path = String.format("%s/system/services/%s/latest-restart", Constants.Gateway.API_VERSION_3,
+                         Constants.Service.APP_FABRIC_HTTP);
+    response = doGet(path);
 
-    Assert.assertEquals(HttpResponseStatus.OK.getCode(), urlConn.getResponseCode());
+    Assert.assertEquals(HttpResponseStatus.OK.getCode(), response.getStatusLine().getStatusCode());
 
     RestartServiceInstancesStatus result =
-      GSON.fromJson(new String(ByteStreams.toByteArray(urlConn.getInputStream()), Charsets.UTF_8),
+      GSON.fromJson(new String(ByteStreams.toByteArray(response.getEntity().getContent()), Charsets.UTF_8),
                     RestartServiceInstancesStatus.class);
 
-    urlConn.disconnect();
+    Assert.assertNotNull(result);
+    Assert.assertEquals(Constants.Service.APP_FABRIC_HTTP, result.getServiceName());
+    Assert.assertEquals(RestartServiceInstancesStatus.RestartStatus.SUCCESS, result.getStatus());
```

Comment:
Ah ok, no it does not.
However since it is internally just delegate to Twill to restart the instances, currently we do not check it. We also do not have tests to test whether add and reduce new instances for service actually restarted.
Looks like good candidate to add those.